### PR TITLE
Topic algorithms - Histogram - J.Y.

### PR DESCRIPTION
**Pull request by LeoYez**
This work is done by Junjie Ye. Student number: 45230095.

### CODE REVIEW

File: image/README.md [hunk 1]
```diff
@@ -1,14 +1,10 @@
-# Imaging
```
Comment (shakes76) [inline image/README.md]: Algorithms files should be in its own folder and not in image folder. Move Readme file. Dependencies list missing

File: image/histogram/histogram.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
+import matplotlib.pyplot as plt
```
Comment (shakes76) [inline image/histogram/histogram.py]: Always good to add a header block. Block would include authorship, license, copyright, dates and description of what the driver does.

File: image/histogram/histogram.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        image_max = tf.reduce_max(image)
+        with tf.Session() as sess:
+            image_min = sess.run(image_min)
+            image_max = sess.run(image_max)
+#     elif source_range == 'dtype':
+#         image_min, image_max = dtype_limits(image, clip_negative=False)
+    image, offset = _offset_array(image, image_min, image_max)#image has already been offset at this moment
+    values = tf.constant(image.ravel())
+    with tf.Session() as sess:
+        values = sess.run(values)
```
Comment (shakes76) [inline image/histogram/histogram.py]: Good to add inline comments even if original code didn't have them

File: image/histogram/main.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from histogram import *
```
Comment (shakes76) [inline image/histogram/main.py]: Always good to add a header block. Block would include authorship, license, copyright, dates and description of what the driver does.

**Comment by shakes76** [comment]
## TF Usage
Adeqaute usage and implementation
## Good Practice (Design/Commenting)
OK, could use more commenting
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present
## Commit Log
Meaningful commit messages 
Progressive commits usage
## Documentation
Description and Comments OK
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
**Feedback  required. Move Readme file into algorithm folder**
Request Description minimal. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

**Comment by LeoYez** [comment]
Sorry for the late update, the readme file has already been moved into the algorithm file.

**Comment by shakes76** [comment]
The images/README.md is gone and there appear to be conflicts. I believe another student has taken the folder name 'histogram' already, rename your folder by adding your name or id to the name and restore the images/README.md.

File: image/README.md (hunk 1 above)
Comment (LeoYez) [reply image/README.md]: Thanks, Shakes. I have had that problem fixed!

**Pull request merged**